drivers/wifi/nrf700x/src/shim.c

```diff
@@ -27,19 +28,73 @@
 #include "qspi_if.h"
 
 LOG_MODULE_REGISTER(wifi_nrf, CONFIG_WIFI_NRF700X_LOG_LEVEL);
+K_HEAP_DEFINE(wifi_drv_mgmt_mem_pool, CONFIG_NRF_WIFI_CONFIG_HEAP_MEM_POOL_SIZE);
+K_HEAP_DEFINE(wifi_drv_pkt_mem_pool, CONFIG_NRF_WIFI_PKT_HEAP_MEM_POOL_SIZE);
 
 struct zep_shim_intr_priv *intr_priv;
 
 static void *zep_shim_mem_alloc(size_t size)
 {
 	size = (size + 4) & 0xfffffffc;
-	return k_malloc(size);
+	return k_heap_alloc(&wifi_drv_mgmt_mem_pool, size, K_FOREVER);
+}
+
+static void *zep_shim_pkt_mem_alloc(size_t size)
+{
+	size = (size + 4) & 0xfffffffc;
+	return k_heap_alloc(&wifi_drv_pkt_mem_pool, size, K_FOREVER);
 }
 
 static void *zep_shim_mem_zalloc(size_t size)
 {
+	void *ret;
+    size_t bounds;
+
+	size = (size + 4) & 0xfffffffc;
+
+	if (size_mul_overflow(size, sizeof(char), &bounds)) {
```

Comment:
is this a new check mandated by new API?

Reply:
This is taken from the checks that were being used in `k_calloc` call.